### PR TITLE
[wip] Mobile touch events

### LINKED ISSUE #657: Support touch events on mobile
Many mobile devices will be able to support mapbox-gl-js. Right now on iOS8, it loads perfectly but the map does not respond to touch events:

cc @mourner

### PR DESCRIPTION
**Pull request by edenh**
Previously https://github.com/mapbox/mapbox-gl-js/pull/749.

A work in progress, resolves #657. Rendering is pretty funny though #750.

- [x] panning
- [x] pinch zoom
- [x] pinch rotate
- [ ] panning while pinch-zooming
- [x] double tap
- [ ] add tests for updated easeTo

### CODE REVIEW

**Comment by alakra** [comment]
Looking forward to this functionality. Thanks for doing this!

**Comment by madjam002** [comment]
I gave this a try on my single page app, it works really well apart from one problem I seemed to be having.
The map I was using it on covered the whole page, but when I called `map.remove()`, I couldn't scroll on any other page on my app even though the map was no longer visible and the map DOM element had been removed. Maybe the touch events aren't being unregistered?

**Comment by 1ec5** [comment]
Might be worth ensuring that how the map responds to touch gestures matches what’s been implemented in Mapbox GL for iOS and Android (mapbox/mapbox-gl-native#1292, mapbox/mapbox-gl-native#553).

**Comment by blq** [comment]
What's the status of touch for the web version? 
I don't get any touch response in Chrome on Sony Z3 or HTC One M8  for example :(

File: js/ui/handlers.js [hunk 1]
```diff
@@ -132,13 +128,27 @@ function Handlers(map) {
...
         })
+        .on('pinchstart', function() {
+            startScale = map.transform.scale;
+            startBearing = map.transform.bearing;
+        })
+        .on('pinch', function(e) {
+            var zoom = map.transform.scaleZoom(startScale * e.scale),
+                bearing = startBearing + e.bearing;
+
+            map.easeTo(null, zoom, bearing, {
```
Comment (tmcw) [inline js/ui/handlers.js]: why not use `undefined` here instead of having add the null check on the other side?

Comment (jfirebaugh) [inline js/ui/handlers.js]: https://github.com/mapbox/mapbox-gl-js/pull/1215 removes the need for `null` or `undefined` positional arguments.

**Comment by vespakoen** [comment]
Hey guys,

Thanks for this! I am able to use mapbox-gl-js on my android phone now because of your great work!

I merged the latest master branch into the touch branch, and only had to change a couple of things afterwards to make it work again (at least on my Samsung Galaxy Mini S4)

I have published it here: https://github.com/trappsnl/mapbox-gl-js/tree/touch

Not sure if I should make a new pull request now? or I guess @edenh can merge my changes into this one?

Anyways, I'll try to help as much as I can for Android support, and have some more devices where I will test this on.

**Comment by StepanKuzmin** [comment]
+1 for this. I'm using `https://mapbox.s3.amazonaws.com/mapbox-gl-js/master/mapbox-gl-dev.js` but touch events still not working (Safari, iPhone 5s).

**Comment by mourner** [comment]
I'll look into finishing and merging this early next week.
@vespakoen thanks for the help!

**Comment by mourner** [comment]
Currently broken because of #1268

**Comment by mourner** [comment]
Will handle subsequent improvements in further commits.

**Comment by vespakoen** [comment]
:+1:  Awesome!! Thanks a lot!

**Pull request merged**